sycl/plugins/rocm/pi_rocm.cpp

```diff
@@ -1639,7 +1655,8 @@ pi_result rocm_piContextCreate(const pi_context_properties *properties,
       break;
     default:
       // Unknown property.
-      assert(!"Unknown piContextCreate property in property list");
+      assert(!static_cast<bool>(
```

Comment:
this doesn't look pretty, and what is the cast of char* to bool doing? also assert will not fire in product (built without assertions)

Reply:
so maybe just use "die" here?

Reply:
Negating char* gives a warning about implicit conversion to bool (regardless of product/debug). So I instead made the cast explicit. I am not sure what do you mean with "die".

Reply:
see line 870 in the same file